packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -204,9 +224,109 @@ const reducer: Reducer<State> = (state = INITIAL_STATE, action) => {
   return state;
 };
 
+const isConfirmReceivedOrNotRequired = async (
+  instance: MongoDBInstance,
+  dataService: DataService,
+  namespace: string,
+  pipeline: Document[]
+) => {
+  const lastStage = pipeline[pipeline.length - 1];
+  const lastStageName = Object.keys(lastStage)[0];
+  let typeOfWrite;
+
+  if (!['$merge', '$out'].includes(lastStageName)) {
+    return true;
+  }
+
+  let { database, collection } = toNS(namespace);
+
+  if (typeof lastStage.$merge === 'string') {
+    collection = lastStage.$merge;
+  }
+
+  if (
+    typeof lastStage.$merge === 'object' &&
+    lastStage.$merge !== null &&
+    typeof lastStage.$merge.into === 'string'
```

Comment:
do we want to handle also ADF here?

https://www.mongodb.com/docs/atlas/data-federation/supported-unsupported/pipeline/merge/

Reply:
We have a bunch of helpers in agg plugin to deal with this (getting namepace, checking if out stage, etc) https://github.com/mongodb-js/compass/blob/a41dd2de57e01ec0ef79f8697be86dc36a299c37/packages/compass-aggregations/src/utils/stage.ts#L111-L132

Reply:
@gribnoysup nice 👍 will refactor with using those!